### PR TITLE
Fix splunksql DB goroutine leak

### LINKED ISSUE #1676: This will cause a goroutine leak
https://github.com/signalfx/splunk-otel-go/blob/a58c95aa48086add859bf20ddcd1cc71e5e30a6a/instrumentation/database/sql/splunksql/sql.go#L60

### LINKED ISSUE #1677: `splunksql`: Not closing the underlying `driver.Connector`
Thesis:

https://github.com/signalfx/splunk-otel-go/blob/a58c95aa48086add859bf20ddcd1cc71e5e30a6a/instrumentation/database/sql/splunksql/sql.go does not close the wrapped db connector (if it implements `io.Closer`).

See: https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/database/sql/sql.go;l=903;drc=1c65b69bd1dbc930c6246877f6c21c81f2a60d55

Would be good to confirm with a unit test.

### PR DESCRIPTION
**Pull request by MrAlias**
fix #1676
fix #1677 

The `Open` function from the `splunksql` instrumentation opens a database to determine the underlying connector or driver. It never closes this database. Update the function that returns the initial database in a way where a user closes the returned database also closes the initial database.

This resolves the issue of the database goroutine being abandoned.

### CODE REVIEW

File: instrumentation/database/sql/splunksql/test/splunksql_test.go [hunk 1]
```diff
@@ -18,9 +18,11 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/suite"
+	"go.uber.org/goleak"
 )
 
 func TestSplunkSQLFullImplementation(t *testing.T) {
+	defer goleak.VerifyNone(t)
```
Comment (pellared) [inline instrumentation/database/sql/splunksql/test/splunksql_test.go:25]: Can you let me know if the test is failing without the `defer db.Close()` in `instrumentation/database/sql/splunksql/sql.go`? 😉

**Review by pellared — APPROVED** [review]
Looks great 👍 

PS. I spent a ~2 hours reviewing this PR.  was adding and later deleting comments a few times 😆

File: instrumentation/database/sql/splunksql/sql.go [hunk 2]
```diff
@@ -90,10 +95,40 @@ type dsnConnector struct {
...
+	if err := c.initDB.Close(); err != nil {
+		return err
+	}
+	if closer, ok := c.Connector.(io.Closer); ok {
+		// This may call the same underlying Connector's close method twice,
+		// the first call coming from the initDB.Close. However, if this isn't
+		// called here and the underlying Connector's close method is different
+		// than the one from initDB we would leave things in an open state.
+		// Better to ensure the operation is performed and assume the
+		// underlying Close is idempotent.
```
[the remark this answers is not in the document]
Comment (MrAlias) [reply instrumentation/database/sql/splunksql/sql.go:130]: Yeah, I was wondering about this as well. I don't think this warrants an option given how unlikely an issue will arise and if a connector returns an error the second time it is closed I expect a user can just ignore that error.

I would classify panicking here as incorrect behavior by the connector and something they need to fix. If that is their behavior, a user can always wrap the close with a recover in the interim.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 This upgrades [OpenTelemetry Go to v1.12.0/v0.35.0][otel-v1.12.0] and
 [OpenTelemetry Go Contrib to v1.13.0/v0.38.0/v0.7.0][contrib-v1.13.0].
 
+### Fixed
+
+- The goroutine in created in by the `Open` function in
```
Comment (pellared) [inline CHANGELOG.md:16]: Is there not too much "in"? 😋 

```suggestion
- The goroutine created by the `Open` function in
```

**Pull request merged**